File: c_solutions_1-10/Euler_9.c
#include <stdio.h>
#include <time.h>
#include <math.h>


int main(void) {

    clock_t start, stop;
    start = clock();

    double a, b;
    long c;
    
    for (a=1, b=2; a <= 3000; a++, b=(a+1)) {
        for (; b < 3000; b++) {
            if (a + b + (sqrt (pow (a, 2) + pow (b, 2))) == 1000) {
                c = a * b * (sqrt (pow (a, 2) + pow (b, 2)));
                goto found_triplet;
            }
            if (a + b + (sqrt (pow (a, 2) + pow (b, 2))) > 1000) 
                break;
        }
    }

    found_triplet:

    stop = clock();
    printf ("Answer: %ld\n", c);
    printf ("Time: %f\n", ((float)stop - (float)start) / CLOCKS_PER_SEC);

    return 0;
}

File: perl_solutions_1-10/Euler_9.pl
use strict;
use warnings;
use Math::Complex;
use Time::HiRes qw( clock );


sub pythag_trip
{
    my ($low, $high) = @_;
    for my $a ($low..$high) {
        for my $b ($a + 1..$high) {
            if (($a + $b + sqrt($a**2 + $b**2)) == 1000) {
                return $a * $b * int(sqrt($a**2 + $b**2));
            }
            elsif (($a + $b + sqrt($a**2 + $b**2)) > 1000) {
                last;
            }
        }
    }
}

my $start = clock();
my $answer = pythag_trip(1, 3000);
my $stop = clock();
print "Answer: ", $answer, "\n";
printf "Time: %.5f\n", $stop - $start;
